pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -175,61 +176,98 @@ func determineSeed(shoot *gardencorev1beta1.Shoot, seedLister gardencorelisters.
 		return nil, err
 	}
 
-	seedsMatchingCloudProfileSelector, err := filterSeedsMatchingSeedSelector(seedList, cloudProfile.Spec.SeedSelector, "CloudProfile")
-	if err != nil {
-		return nil, err
+	filteredSeeds, err := filterUsableSeeds(seedList)
+	if err == nil {
+		filteredSeeds, err = filterSeedsMatchingSeedSelector(filteredSeeds, cloudProfile.Spec.SeedSelector, "CloudProfile")
 	}
-	seedsMatchingShootSelector, err := filterSeedsMatchingSeedSelector(seedsMatchingCloudProfileSelector, shoot.Spec.SeedSelector, "Shoot")
-	if err != nil {
-		return nil, err
+	if err == nil {
+		filteredSeeds, err = filterSeedsMatchingSeedSelector(filteredSeeds, shoot.Spec.SeedSelector, "Shoot")
 	}
-
-	candidates, err := getCandidates(shoot, seedsMatchingShootSelector, strategy)
-	if err != nil {
-		return nil, err
+	if err == nil {
+		filteredSeeds, err = filterSeedsMatchingProviders(cloudProfile, shoot, filteredSeeds)
+	}
+	if err == nil {
+		filteredSeeds, err = filterCandidates(shoot, filteredSeeds)
 	}
+	if err == nil {
+		filteredSeeds, err = applyStrategy(shoot, filteredSeeds, strategy)
+	}
+	if err == nil {
+		return getSeedWithLeastShootsDeployed(filteredSeeds, shootList)
+	}
+	return nil, err
+}
 
-	filteredCandidates, err := filterCandidates(shoot, candidates)
-	if err != nil {
-		return nil, err
+func isUsableSeed(seed *gardencorev1beta1.Seed) bool {
+	return seed.DeletionTimestamp == nil && seed.Spec.Settings.Scheduling.Visible && common.VerifySeedReadiness(seed)
+}
+
+func filterUsableSeeds(seedList []*gardencorev1beta1.Seed) ([]*gardencorev1beta1.Seed, error) {
+	var matchingSeeds []*gardencorev1beta1.Seed
+
+	for _, seed := range seedList {
+		if isUsableSeed(seed) {
+			matchingSeeds = append(matchingSeeds, seed)
+		}
 	}
 
-	return getSeedWithLeastShootsDeployed(filteredCandidates, shootList)
+	if len(matchingSeeds) == 0 {
+		return nil, fmt.Errorf("none of the %d seeds is valid for scheduling (not deleting, visible and ready)", len(seedList))
+	}
+	return matchingSeeds, nil
 }
 
-func filterSeedsMatchingSeedSelector(seedList []*gardencorev1beta1.Seed, labelSelector *metav1.LabelSelector, kind string) ([]*gardencorev1beta1.Seed, error) {
-	selector := &metav1.LabelSelector{}
-	if labelSelector != nil {
-		selector = labelSelector
+func filterSeedsMatchingSeedSelector(seedList []*gardencorev1beta1.Seed, seedSelector *gardencorev1beta1.SeedSelector, kind string) ([]*gardencorev1beta1.Seed, error) {
```

Comment:
```suggestion
func filterSeedsMatchingLabelSelector(seedList []*gardencorev1beta1.Seed, seedSelector *gardencorev1beta1.SeedSelector, kind string) ([]*gardencorev1beta1.Seed, error) {
```

would prefer that name as this check only uses the label selector of the provided Seed selector.